Makefile

```diff
@@ -88,14 +88,14 @@ uninstall:
 		$(DESTDIR)$(PREFIX)/bin/keyd-usb-gadget.sh \
 		$(DESTDIR)$(PREFIX)/lib/systemd/system/keyd.service
 clean:
-	-rm -rf bin keyd.service
+	rm -rf bin data/*.1.gz data/keyd.compose keyd.service
```

Comment:
compose and run are deliberately not included the default/clean targets so the user doesn't need to have python or scdoc on their system. The generated assets are deliberately checked into the repo.

Reply:
Pros to removing generated files:
  * It simplifies the work of downstream packagers that have to tweak the input files or the build process.
  * It eliminates any chance that the checked-in generated files become out of sync with the checked-in input files.
  * The codebase is easier for potential contributors to understand.
  * No need to review changes to generated files when a pull request modifies the input files.

Cons to removing generated files:
  * Users building keyd will have to install python and scdoc.  (Users running a pre-built binary package are unaffected.)  Such users already have to know how to install a C compiler and dependency headers; adding python and scdoc to the list does not seem overly burdensome to me.
  * Building takes much more time now.  (It's still sufficiently fast enough, in my opinion.  Also, incremental build speed can be restored with some further tweaks to the `Makefile`.)